src/hooks/useCheckImage.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+
+export const useCheckImage = (
+  src: string
+): [boolean, boolean, string | undefined] => {
+  const [isLoading, setLoading] = useState(false);
+  const [error, setError] = useState<string>();
+
+  useEffect(() => {
+    const checkImage = new Image();
+    checkImage.src = src;
+
+    setLoading(true);
+    checkImage
+      .decode()
+      .then(() => {
+        setError(undefined);
+        setLoading(false);
+      })
+      .catch((err) => {
+        setError(err.message);
+        setLoading(false);
+      });
+  }, [src]);
+
+  return [!error, isLoading, error];
```

Comment:
It would be better if you change this to be an object, that way the order of imports won't matter, and you can import the first and the last one.

Reply:
Generally speaking, I'd argue that using tuples as return type from hooks is more _convenient_ as 

- you don't have to rename returned fields every time the hook is used which results in more concise usage
- the practice with tuples is widely used in react and other hooks libraries

But here we are returning 3 values and two of them are of similar type `boolean`. So I changed it to object.
Commit https://github.com/sparkletown/sparkle/commit/f033a214d20db26a1e8d035a115ccd345272d54f